bin/server.js

```diff
@@ -41,35 +42,49 @@ async function startServer () {
   app.use(cors({ origin: '*' }))
 
   // MIDDLEWARE END
-
   // Retrieve hash from BCH network and retrieve data from IPFS.
 
   // Get the latest hash off the BCH network.
   const hash = await bch.findHash()
+  // To-Do: Add some retry code here. If it doesn't find the hash on the first try,
+  // it should retry a few times. The npm p-retry library would be good to use here.
 
   // Exit if no hash is found.
   if (!hash) {
-    console.log(`Could not find IPFS hash associated with BCH address ${config.BCHADDR}`)
-    console.log(`Publish an IPFS hash using the memo-push tool before running this server.`)
-    // console.log(`Exiting`)
-    // process.exit()
+    console.log(
+      `Could not find IPFS hash associated with BCH address ${config.BCHADDR}`
+    )
+    console.log(
+      `Publish an IPFS hash using the memo-push tool before running this server.`
+    )
+    console.log(`Exiting`)
+    process.exit()
   }
 
-  console.log(`Retrieving and serving this IPFS hash: ${hash}`)
+  // Start IPFS
+  const ipfsNode = await ipfs.startIPFS()
+
+  // For ipfs Ready
+  ipfsNode.on('ready', async () => {
+    console.log(`Retrieving and serving this IPFS hash: ${hash}`)
 
-  // Get the content from the IPFS network.
-  shell.cd(`ipfs-data`)
-  shell.exec(`ipfs get ${hash}`)
-  shell.exec(`ipfs pin add ${hash}`)
+    // Bootstrap connection (for development purposes)
+    // Replace the multiaddr below with the IPFS node running on your local machine.
+    await ipfsNode.swarm.connect('/ip4/127.0.0.1/tcp/4001/ipfs/QmSgDzV1GeTg1tx4wU5WKjxMvT692xvt8FS14JdoDEgFjj')
 
-  // Mount the downloaded directory and serve it.
-  app.use(convert(mount('/', serve(`${process.cwd()}/${hash}`))))
+    // Get the latest content from the IPFS network and Add into ipfs-data.
+    await ipfs.getContent(ipfsNode, hash)
+    // Adds an IPFS object to the pinset and also stores it to the IPFS repo.
+    ipfs.pinAdd(ipfsNode, hash)
 
-  await app.listen(config.port)
-  console.log(`Server started on ${config.port}`)
-  console.log(`Access website at http://localhost:${config.port}/`)
+    // Mount the downloaded directory and serve it.
+    app.use(convert(mount('/', serve(`${process.cwd()}/ipfs-data/${hash}`))))
```

Comment:
You asked about this line. This code instructs Koa to mount the directory after IPFS has downloaded the data. It mounts it to the root path `/`. Basically, this is how Koa serves the website after its been downloaded.